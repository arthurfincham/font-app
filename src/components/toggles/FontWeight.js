import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
import { styled } from '@mui/system';
import { Typography } from '@mui/material';
import MyTooltip from '../MyTooltip';
export default function FontWeight({ fontWeight, setFontWeight }) {
  const StyledToggleButtonGroup = styled(ToggleButtonGroup)(({ theme }) => ({
    '& .MuiToggleButtonGroup-grouped': {
      margin: '5px 0px ',
    },
  }));
  const handleWeight = (event, newAlignment) => {
    setFontWeight(newAlignment);
  };
  return (
    <MyTooltip title="Weight">
      <StyledToggleButtonGroup size="small" value={fontWeight} exclusive onChange={handleWeight} aria-label="text alignment">
        <ToggleButton value="400" aria-label="left aligned">
          <Typography variant="button" sx={{ fontWeight: 'bold' }}>
            400
          </Typography>
        </ToggleButton>
        <ToggleButton value="500" aria-label="left aligned">
          <Typography variant="button" sx={{ fontWeight: 'bold' }}>
            500
          </Typography>
        </ToggleButton>
        <ToggleButton value="600" aria-label="left aligned">
          <Typography variant="button" sx={{ fontWeight: 'bold' }}>
            600
          </Typography>
        </ToggleButton>
        <ToggleButton value="700" aria-label="left aligned">
          <Typography variant="button" sx={{ fontWeight: 'bold' }}>
            700
          </Typography>
        </ToggleButton>
        <ToggleButton value="800" aria-label="left aligned">
          <Typography variant="button" sx={{ fontWeight: 'bold' }}>
            800
          </Typography>
        </ToggleButton>
      </StyledToggleButtonGroup>
    </MyTooltip>
  );
}
