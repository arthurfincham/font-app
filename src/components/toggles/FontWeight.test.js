import FontWeight from './FontWeight';
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

it('it correctly renders the FS toggle', () => {
  const wrapper = shallow(<FontWeight />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('has three toggles', () => {
  const wrapper = mount(<FontWeight />);
  expect(wrapper.find('button')).toHaveLength(5);
});
