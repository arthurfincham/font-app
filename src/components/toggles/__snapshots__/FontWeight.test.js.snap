// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<ToggleTooltip
  title="Weight"
>
  <Styled(ForwardRef(ToggleButtonGroup))
    aria-label="text alignment"
    exclusive={true}
    onChange={[Function]}
    size="small"
  >
    <ForwardRef(ToggleButton)
      aria-label="Set Weight 400"
      key="400"
      value="400"
    >
      <ForwardRef(Typography)
        sx={
          Object {
            "fontWeight": "bold",
          }
        }
        variant="button"
      >
        400
      </ForwardRef(Typography)>
    </ForwardRef(ToggleButton)>
    <ForwardRef(ToggleButton)
      aria-label="Set Weight 500"
      key="500"
      value="500"
    >
      <ForwardRef(Typography)
        sx={
          Object {
            "fontWeight": "bold",
          }
        }
        variant="button"
      >
        500
      </ForwardRef(Typography)>
    </ForwardRef(ToggleButton)>
    <ForwardRef(ToggleButton)
      aria-label="Set Weight 600"
      key="600"
      value="600"
    >
      <ForwardRef(Typography)
        sx={
          Object {
            "fontWeight": "bold",
          }
        }
        variant="button"
      >
        600
      </ForwardRef(Typography)>
    </ForwardRef(ToggleButton)>
    <ForwardRef(ToggleButton)
      aria-label="Set Weight 700"
      key="700"
      value="700"
    >
      <ForwardRef(Typography)
        sx={
          Object {
            "fontWeight": "bold",
          }
        }
        variant="button"
      >
        700
      </ForwardRef(Typography)>
    </ForwardRef(ToggleButton)>
    <ForwardRef(ToggleButton)
      aria-label="Set Weight 800"
      key="800"
      value="800"
    >
      <ForwardRef(Typography)
        sx={
          Object {
            "fontWeight": "bold",
          }
        }
        variant="button"
      >
        800
      </ForwardRef(Typography)>
    </ForwardRef(ToggleButton)>
  </Styled(ForwardRef(ToggleButtonGroup))>
</ToggleTooltip>
`;
